### PR TITLE
Rewrite transaction test

### PR DESCRIPTION
**Pull request by hououou**
- Expand multiple connections to support default connection if not create connection previously
  -use case:
    -CREATE_CONNECTION conn_write
     -STATEMENT [conn_write] BEGIN WRITE TRANSACTION
     ---- ok
     -STATEMENT BEGIN READ TRANSACTION
     ---- ok

- Expand the test framework to support batch statements; the file is loaded from KUZU_ROOT_PATH/test/statements
 -use case:  
 -BATCH_STATEMENTS \<FILE\>:xx.cypher
 ---- ok

- Rewrite the following tests to test_file
  - e2e_create_macro_test.cpp
  - e2e_create_rel_test.cpp
  - e2e_delete_create_transaction_test.cpp
   - e2e_delete_rel_test.cpp
   - e2e_update_rel.cpp

### CODE REVIEW

File: test/test_files/transaction/create_rel/insert_rels_to_small_list.test [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+-STATEMENT MATCH (a:person)-[e:knows]->(b:person) where b.ID > 200 AND a.ID = 1 RETURN e.length, e.place, e.tag
+---- 4
+300|this is a long str|[123,good]
+34|short|
+45||[123,good]
+||
+
+
+
+#insertRelsToSmallListRollbackNormalExecution
```
Comment (andyfengHKU) [inline test/test_files/transaction/create_rel/insert_rels_to_small_list.test:56]: Let's remove this comment for all tests. They seem to be duplicate with the CASE name.

Also I think we probably also don't need LOG after CASE. LOG is useful when we try to differentiate multiple statements in a single CASE (which is more common in read only tests). @ray6080 can comment this

File: test/test_files/transaction/create_rel/violate_error.test [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+#ViolateOneOneMultiplicityError
+-CASE ViolateOneOneMultiplicityError
+-LOG ViolateOneOneMultiplicityError
+-STATEMENT BEGIN WRITE TRANSACTION
+---- ok
+-STATEMENT MATCH (a:animal), (p:person) WHERE a.ID = 2 AND p.ID = 10 CREATE (a)-[:hasOwner]->(p);
+---- error
+Runtime exception: Node in RelTable 3 cannot have more than one neighbour in the forward direction.
```
Comment (andyfengHKU) [inline test/test_files/transaction/create_rel/violate_error.test:22]: New line. Don't forget to check other files too.

File: test/test_runner/test_runner.cpp [hunk 3]
```diff
@@ -5,6 +5,7 @@
 #include "common/exception/test.h"
 #include "common/string_utils.h"
 #include "spdlog/spdlog.h"
+#include "test_helper/test_helper.h"
```
Comment (andyfengHKU) [inline test/test_runner/test_runner.cpp:8]: this must be a duplicate include

File: test/include/graph_test/graph_test.h [hunk 4]
```diff
@@ -169,19 +169,26 @@ class DBTest : public BaseGraphTest {
...
                 createConns(connNames);
                 continue;
             }
-            if (conn) {
+            if (conn) { // for normal conn
                 TestRunner::runTest(statement.get(), *conn, databasePath);
-            } else {
-                auto conn_name = *statement->conn_name;
+            } else { // for multiple conn
+                auto conn_name = *statement->connName;
```
Comment (andyfengHKU) [inline test/include/graph_test/graph_test.h:180]: connName

File: test/test_runner/test_parser.cpp [hunk 5]
```diff
@@ -351,18 +353,24 @@ void TestParser::tokenize() {
...
         if (matches.size() == 3) {
             statement_status = true;
-            statement->conn_name = matches[1];
+            statement->connName = matches[1];
             query = matches[2];
         }
     }
-    return statement_status;
+
+    auto& caseStatus = testGroup->multipleConns;
```
Comment (andyfengHKU) [inline test/test_runner/test_parser.cpp:368]: So you are assigning default connection name when `multipleConn` is enabled and a `connName` can not be found through regex matching. I wonder if we can simplify the logic as 

```
if connName not found through regex
     statement->connName = DEFAULT
```

**Review by andyfengHKU — COMMENTED** [review]
Let's remove all cpp tests that have been converted

File: test/include/graph_test/graph_test.h [hunk 6]
```diff
@@ -169,19 +169,26 @@ class DBTest : public BaseGraphTest {
...
         std::set<std::string> connNames = std::set<std::string>()) {
         for (auto& statement : statements) {
-            if (statement->reLoadDBFlag) {
+            if (statement->reloadDBFlag) {
                 createDB(checkpointWaitTimeout);
                 createConns(connNames);
                 continue;
             }
-            if (conn) {
+            if (conn) { // for normal conn
```
Comment (ray6080) [inline test/include/graph_test/graph_test.h:177]: Remove comment

File: test/include/graph_test/graph_test.h [hunk 7]
```diff
@@ -169,19 +169,26 @@ class DBTest : public BaseGraphTest {
...
                 TestRunner::runTest(statement.get(), *conn, databasePath);
-            } else {
-                auto conn_name = *statement->conn_name;
+            } else { // for multiple conn
```
Comment (ray6080) [inline test/include/graph_test/graph_test.h:179]: Remove comment

File: test/include/test_runner/test_group.h [hunk 8]
```diff
@@ -21,8 +21,9 @@ struct TestStatement {
     bool checkOutputOrder = false;
     std::string expectedTuplesCSVFile;
     // for multiple conns
-    std::optional<std::string> conn_name; // default conntion name conn0
-    bool reLoadDBFlag = false;
+    std::string batchStatmentsCSVFile;
+    std::optional<std::string> connName; // default conntion name conn0
```
Comment (ray6080) [inline test/include/test_runner/test_group.h:25]: Remove comment.

**Review by ray6080 — APPROVED** [review]
Address comments and rebase before you merge.
Also, update our documentation on testing framework after you merge the PR.

File: test/test_runner/test_runner.cpp (hunk 3 above)
Comment (hououou) [reply test/test_runner/test_runner.cpp:8]: it needs to be included

**Pull request merged**